packages/backend/src/core/DiscoveryWatcher.ts

```diff
@@ -48,13 +43,15 @@ export class DiscoveryWatcher {
   }
 
   async update() {
+    const metricsDone = initMetrics()
```

Comment:
Is this approach necessary? We could just start the timer here and stop it at the end. I'm not sure, but I have to admit I actually quite like how it looks here.